utils/core/base_xtest.py

```diff
@@ -22,6 +22,10 @@ def defaultMain(argv, runDocs, runTests):
 
 
 class BaseTest(BaseTestRunner):
+    def __init__(self):
```

Comment:
I'm afraid this BaseTest extension , however excelent idea, should be in separate commit.

Reply:
The reason of the refactoring was the main purpose of this PR -> i dont see it as a refactoring and a change, its just a bigger change, where neccesary part of it was the refactoring. In the future, there wont be a time when you will want to revert just to the refactoring in this commit as it would prove useless. And the fix would be nonfunctional without the refactoring. Plus the change should have been there a long time ago, but only now I spotted this problem. 
I really dont see the need for splitting this into two commits other then forcing our intern into practicing git skills.

Reply:
To have refactoring separate from logic changes is mandatory practice.  It allows to see the logic change clear, and is really really good habit to get.
If it is  - as you are saying - inseparable change then tehre is something wrong. It may be my lack of python knowlege. Can  you please elaborate more?